generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FixedFieldBuilder.java

```diff
@@ -34,13 +34,6 @@ public FixedFieldBuilder(
     public FixedField findNextFixedField(ReductiveState reductiveState, ConstraintNode rootNode, FixFieldStrategy fixFieldStrategy) {
         Field fieldToFix = fixFieldStrategy.getNextFieldToFix(reductiveState, rootNode);
 
-        if (fieldToFix == null){
```

Comment:
Is this removal correct; what if the strategy cannot determine a field to fix next - for some reason, you'd get a NullPointer further down the chain - i'd suggest keeping it?

Reply:
It's an interesting question.
IMO you should avoid null checks unless you know it could be null, and also what to do in that circumstance.

A point could be made everywhere that a null could happen due to some error or another, but we don't have random null checks in other parts of the code.

i'm leaning against putting it back, mostly because it doesn't fix anything, the system will still error at a similar place, and we have no reason to believe this will ever happen